Adding international shipping option

iAlgebra opened the pull request:
![667235f11c467c2c3298af1a2de16930](https://user-images.githubusercontent.com/16955860/39734669-af4420e4-52bb-11e8-9bda-fca410869b9e.png)

The "International Shipping" checkbox was added.
The "Subtotal" input was created to display the subtotal amount without the shipping charges.
The "Shipping" input was created to showcase the shipping fee.

iAlgebra committed:
Adding international shipping option

The "International Shipping" checkbox was added.
The "Subtotal" input was created to display the subtotal amount without the shipping charges.
The "Shipping" input was created to showcase the shipping fee.

fsodano commented on book.html:
```diff
@@ -22,15 +22,34 @@ <h2>Author: Marco Antonio Solis</h2>
                 swine venison jerky fatback. Shoulder swine shankle fatback pork. Chicken pastrami landjaeger hamburger
                 shoulder. Bacon spare ribs filet mignon shank bresaola beef ribs pancetta sausage pork belly drumstick
-                frankfurter.
+                frankfurter.</p>
```
This should be in the next line

fsodano commented on book.html:
```diff
@@ -22,15 +22,34 @@ <h2>Author: Marco Antonio Solis</h2>
+
+                <div class="input-group">
+                    <label class="input-group-addon" for="shipping">Shipping</label>
+                    <input type="number" min="0" step="0.01" class="form-input" id="shipping" disabled value="0"
```
In the previous input you've added `<input type="number" min="0" step="0.01" class="form-input" id="subtotal" value="0" disabled/>`, in this input you've placed first the `value` attribute and then the `disabled` one. Can you keep it consistent please?

fsodano commented on book.js:
```diff
@@ -1,7 +1,11 @@
+var intShippingPrice = Number($shipping.dataset.intShipping);
+var $total = document.getElementById('total');
 
 var maxQuantity = 100;
```
Not for this PR, but I think this could be read from a data- attribute.

fsodano commented on book.js:
```diff
@@ -15,7 +19,19 @@ for (var i = 0; i < maxQuantity; i++) {
+  if ($shippingCheckbox.checked) {
+    $shipping.value = intShippingPrice;
+  } else {
+    $shipping.value = 0;
```
You are defaulting to 0, but maybe the default shipping can change in the future. I think you should read this value from a `data-` attribute like you're doing with the `intShippingPrice`. Perhaps you can add on the same element, a `data-base-shipping` data attribute.

fsodano commented on book.js:
```diff
@@ -15,7 +19,19 @@ for (var i = 0; i < maxQuantity; i++) {
+  $total.value = Number($subtotal.value) + Number($shipping.value);
+};
+
+$total.value = Number($subtotal.value) + Number($shipping.value);
```
You are repeating this line 3 times. I think you can put this in a function called `updateTotalValue()` and then you just call that function multiple times.

```javascript
function updateTotal(){
    $total.value = Number($subtotal.value) + Number($shipping.value);
}
```

iAlgebra committed:
Formatting book paragraph

iAlgebra committed:
Removing hardcoded max quantity

Previously, the max quantity of books that could be ordered was hardcoded in JS, now this value is being read from a data attribute.

iAlgebra committed:
Adding default value for Total field

iAlgebra committed:
Reading base shipping price from a data attribute

Previously, the base shipping price was hardcoded to be 0

iAlgebra committed:
Calculating Total value

Every time the user selects a quantity or selects an international shipping option, the total value is updated. Subtotal and Total values are also formatted to 2 decimal places.

fsodano reviewed (approved):
🥇 👏 👏 👏
